test/ecs/customComponentWithSystems.spec.ts

```diff
@@ -294,37 +302,123 @@ describe('test CRDT flow E2E', () => {
 
       // We send the message from B -> A
       await engineB.update(0)
-      expect(env.interceptedMessages).toMatchObject([
+      expect(env.connection.interceptedMessages).toMatchObject([
         // this value will have has the same timestamp in both engines
         {
           direction: 'b->a',
           componentId: 123987,
-          entity: entityA,
+          entityId: entityA,
           data: Uint8Array.of(88),
           timestamp: 6
         }
       ])
-      env.interceptedMessages.length = 0
+      env.connection.interceptedMessages = []
     })
 
     it('now we are receiving the updates from engineB', async () => {
       // run update tick on engineA so we receive the message of the component that we already remove
       await engineA.update(0)
-      expect(env.interceptedMessages).toMatchObject([
+
+      // The conflict here is with same timestamp, `engineA` has removed the component, and `engineB` modified it.
+      //  The "greater" data wins here, so the state converges to `engineB` LWW.
+
+      expect(env.connection.interceptedMessages).toMatchObject([])
+
+      await Promise.all([engineA.update(0), engineB.update(0)])
+      expect(env.connection.interceptedMessages).toMatchObject([])
+
+      expect(int8A.getOrNull(entityA)).toBe(88)
+      expect(int8B.getOrNull(entityA)).toBe(88)
+    })
+  })
+  // describe('conflict resolution case 4', () => {
+  //   it('same as case 3) but with roles inverted', async () => {
```

Comment:
Is there a reason to be commented instead of .skip?

Reply:
nope, thanks for the catch, I'm restoring that lines